static/js/portico/signup.js

```diff
@@ -136,14 +136,26 @@ $(() => {
         }
     });
 
-    const show_subdomain_section = function (bool) {
-        const action = bool ? "hide" : "show";
-        $("#subdomain_section")[action]();
+    const subdomain_required = function (bool) {
+        // Required attribute of subdomain field should be toggled only when
+        // radio button choices are present, i.e., root domain is available.
+        if ($("#realm_with_subdomain").length > 0) {
+            if (bool) {
+                $("#id_team_subdomain").prop("required", true);
+                $("#id_team_subdomain").addClass("required");
+                $("#id_team_subdomain-error").show();
+            } else {
+                $("#id_team_subdomain").prop("required", false);
+                $("#id_team_subdomain").removeClass("required");
+                $("#id_team_subdomain-error").hide();
+            }
+        }
     };
 
-    $("#realm_in_root_domain").on("change", function () {
-        show_subdomain_section($(this).is(":checked"));
+    $("input[name='realm_url']:radio").on("change", () => {
+        subdomain_required($("#realm_with_subdomain").is(":checked"));
```

Comment:
The `subdomain_required` parameter seems superfluous. We can just move `$("#realm_with_subdomain").is(":checked")` into `subdomain_required`.

Reply:
Updated.